### DISCUSSION TITLE
Is there any source for dev-icons

### DISCUSSION THREAD

**Discussion by sriramkandukuri** [hrsh7th/nvim-cmp#702] [Q&A] [answered]
Hi,

dimtmel/cmp-digraphs do upto digraphs only,
Is there any source for dev-icons and unicode characters with completion

is there any alternative for unicode.vim written in lua with nvim-cmp as backend?

Thanks,
Sriram

**Comment by dmitmel** [comment, marked as answer]
> Is there any source for dev-icons and unicode characters with completion

It's pretty easy to create a source for this, here you go:

```lua
local devicons = require('nvim-web-devicons')
cmp.register_source('devicons', {
  complete = function(self, params, callback)
    local items = {}
    for _, icon in pairs(devicons.get_icons()) do
      table.insert(items, {
        label = icon.icon .. '  ' .. icon.name,
        insertText = icon.icon,
        filterText = icon.name,
      })
    end
    callback({ items = items })
  end,
})
```

> is there any alternative for unicode.vim written in lua with nvim-cmp as backend?

Not that I know of any, but this isn't hard to do either - the only thing you really need is a database of all Unicode characters with their names and codepoints, formatted in a way that is easy to read from Lua (JSON, for example).

**Reply by sriramkandukuri** [reply]
Thanks for suggestion

**Reply by zackfall** [reply]
where should i place that? xdd i place it in the init.lua but no icon appear in the completions
